git worktree support

By using jgit from eclipse this plugin will fail when used within a git worktree.

https://bugs.eclipse.org/bugs/show_bug.cgi?id=477475
possible fix:
https://git.eclipse.org/r/#/c/54404/
https://dev.eclipse.org/mhonarc/lists/jgit-dev/msg02987.html

The issue is quite simple to recreate, just checkout a normal project that uses this plugin and then create a worktree copy with `git worktree add ../other master` (you may have to use a different branch since you can only have one checkout of a branch). Then go to `../other` and execute the build, it should throw something like this:

```
Caused by: org.eclipse.jgit.api.errors.NoHeadException: No HEAD exists and no explicit starting revision was specified
        at org.eclipse.jgit.api.LogCommand.call(LogCommand.java:150)
        at org.eclipse.jgit.api.LogCommand.call(LogCommand.java:1)
        at java_util_concurrent_Callable$call$0.call(Unknown Source)
        at org.ajoberstar.grgit.operation.LogOp.call(LogOp.groovy:64)
        at org.ajoberstar.grgit.operation.LogOp.call(LogOp.groovy)
        at java_util_concurrent_Callable$call.call(Unknown Source)
        at java_util_concurrent_Callable$call.call(Unknown Source)
        at org.ajoberstar.grgit.internal.OpSyntax.mapOperation(OpSyntax.groovy:20)
        at org.ajoberstar.grgit.internal.OpSyntax$mapOperation.callStatic(Unknown Source)
        at org.ajoberstar.grgit.Grgit.log(Grgit.groovy)
        at org.ajoberstar.grgit.Grgit$log$0.call(Unknown Source)
        at net.nemerosa.versioning.git.GitInfoService.getInfo(GitInfoService.groovy:53)
        at net.nemerosa.versioning.SCMInfoService$getInfo.call(Unknown Source)
        at net.nemerosa.versioning.VersioningExtension.computeInfo(VersioningExtension.groovy:195)
        at net.nemerosa.versioning.VersioningExtension.getInfo(VersioningExtension.groovy:182)
        at net.nemerosa.versioning.VersioningExtension_Decorated.getInfo(Unknown Source)
```

I encountered it while working on https://github.com/spockframework/spock 

JGit still doesn't have a fix it seems: https://git.eclipse.org/r/#/c/126454/
However, you could add a workaround and check if `.git` is a file instead of a directory and then just inject a dummy version instance and log a warning. IMHO this is a good compromise, to be able to use worktrees, since you won't use them to release the dummy version shouldn't be such a problem.

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -12,7 +12,7 @@ plugins {
     /**
      * Versioning applied to itself
      */
-    id("net.nemerosa.versioning") version "2.14.0"
+    id("net.nemerosa.versioning") version "3.1.0"
     /**
      * Release in GitHub
      */

--- a/gradle.properties
+++ b/gradle.properties
@@ -1,5 +1,5 @@
 # Some dependency versions
-jgitVersion = 6.1.0.202203080745-r
+jgitVersion = 7.6.0.202603022253-r
 
 # Placeholders for Bintray credentials
 bintrayUser =

--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.3-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.3-bin.zip
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists

